File: teams/mjg-clojure/brisfunctional/src/brisfunctional/problem1.clj
(ns brisfunctional.problem1)

(defn sum-of-multiples-of-3-or-5-below [n]
  
  (reduce +
    (filter 
      #(or
         (= 0 (mod % 3))
         (= 0 (mod % 5))) 
      (range 1 n))))

File: teams/mjg-haskell/BrisFunctional/Problem1.hs
module BrisFunctional.Problem1
( problem1
, testProblem1 )
where

import Test.HUnit

-- Problem 1
-- Find the sum of all the multiples of 3 or 5 below 1000.
problem1 :: (Integral a) => a
problem1 = sum [ x | x <- [1..999], or [(x `mod` 3 == 0), (x `mod` 5 == 0)] ]

-- test case
testProblem1 = TestCase $ assertEqual 
  "Problem 1 answer = 233168"
  233168 
  ( problem1 ) 

